### PR TITLE
Endpoint to list all source systems

### PR DESCRIPTION
**Pull request by southeo**
New endpoint to list all source systems (paginated). New endpoint returns results according to JsonApi standard 
Added tests for Source System Data flow (controller, service, and repository)

### CODE REVIEW

File: src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java [hunk 1]
```diff
@@ -30,4 +38,20 @@ public SourceSystemRecord updateSourceSystem(String id, SourceSystem sourceSyste
     repository.updateSourceSystem(sourceSystemRecord);
     return sourceSystemRecord;
   }
+
+  public JsonApiWrapper getSourceSystemRecords(int pageNum, int pageSize, String path){
+    var ssRecords = repository.getSourceSystems(pageNum, pageSize);
+    var dataNode = wrapData(ssRecords);
+    return new JsonApiWrapper(dataNode, new JsonApiLinks(path));
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java:45]: Does this comply to the pagination specification in JSON:API?
https://jsonapi.org/format/#fetching-pagination
I think there should be link to next (if applicable) and previous (if applicable) 
`last` we discussed before and we ended that for the last page we would not show `next` if there is no next page

File: src/main/java/eu/dissco/orchestration/backend/repository/SourceSystemRepository.java [hunk 2]
```diff
@@ -14,14 +18,12 @@ public class SourceSystemRepository {
   private final DSLContext context;
 
   public int createSourceSystem(SourceSystemRecord sourceSystemRecord) {
-    return context.insertInto(NEW_SOURCE_SYSTEM)
-        .set(NEW_SOURCE_SYSTEM.ID, sourceSystemRecord.id())
+    return context.insertInto(NEW_SOURCE_SYSTEM).set(NEW_SOURCE_SYSTEM.ID, sourceSystemRecord.id())
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/repository/SourceSystemRepository.java:21]: I don't think this has been changed.
Could you revert the change here and keep the indentation as it was.
For readability is it nicer if all the `.set` are on a newline

File: src/main/java/eu/dissco/orchestration/backend/repository/SourceSystemRepository.java [hunk 3]
```diff
@@ -31,7 +33,38 @@ public int updateSourceSystem(SourceSystemRecord sourceSystemRecord) {
...
+    return context.select(NEW_SOURCE_SYSTEM.ID, NEW_SOURCE_SYSTEM.CREATED, NEW_SOURCE_SYSTEM.NAME,
+            NEW_SOURCE_SYSTEM.ENDPOINT, NEW_SOURCE_SYSTEM.DESCRIPTION, NEW_SOURCE_SYSTEM.MAPPING_ID)
+        .from(NEW_SOURCE_SYSTEM)
+        .limit(pageSize)
+        .offset(offset)
+        .fetch(this::mapToSourceSystemRecord);
+  }
+
+  private SourceSystemRecord mapToSourceSystemRecord(
+      Record6<String, Instant, String, String, String, String> row) {
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/repository/SourceSystemRepository.java:50]: You could also just use the interface `Record` here

File: src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java [hunk 4]
```diff
@@ -30,4 +38,20 @@ public SourceSystemRecord updateSourceSystem(String id, SourceSystem sourceSyste
...
+  }
+
+  List<JsonApiData> wrapData(List<SourceSystemRecord> ssRecords){
+    List<JsonApiData> dataNode = new ArrayList<>();
+
+    ssRecords.forEach(ss -> dataNode.add(new JsonApiData(ss.id(), HandleType.SOURCE_SYSTEM, mapper.valueToTree(ss))));
```
Comment (samleeflang) [inline src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java:51]: Another option might be to stream this, 
ssRecords.stream().map().toList()
Then we wouldn't have to create an empty list first and then populate it

**Review by samleeflang — COMMENTED** [review]
Nice small PR! :tada:

File: src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java (hunk 1 above)
Comment (southeo) [reply src/main/java/eu/dissco/orchestration/backend/service/SourceSystemService.java:45]: Oh good catch, linksNode has self, first, and next and prev if applicable

**Review by samleeflang — APPROVED** [review]
Looks good!
